examples/Tutorial.ipynb

```diff
@@ -0,0 +1,925 @@
+    "\n",
+    "For instance, models may be any of the following types currently:\n",
+    "* Gaussian\n",
+    "* Moffat\n",
+    "* Kolmogorov\n",
+    "* PixelGrid\n",
+    "* Optical\n",
+    "\n",
+    "The currently implemented interpolation schemes are:\n",
+    "* Mean\n",
+    "* Polynomial\n",
+    "* BasisPolynomial\n",
+    "* kNNInterp\n",
+    "* GPInterp\n",
+    "\n",
+    "This structure makes all parts of the code very modular.  It is easy to add a new type for some part of the calculation, which a user can then specify in their config file to change the behavior of Piff for their use case."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "5a6c0283",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Here is a relatively simple configuration for DES Y6 data:\n",
+    "\n",
+    "import yaml\n",
+    "\n",
+    "config = \"\"\"\n",
+    "\n",
+    "input:\n",
+    "    \n",
+    "    # dir is optional.  If set, it applies to all file names here without an absolute path\n",
+    "    dir: 572501\n",
+    "        \n",
+    "    # Define the input image with the pixel data.\n",
+    "    # These data come with a bad pixel mask and a weight map.\n",
+    "    image_file_name: D00572501_z_c01_r3624p01_immasked.fits.fz\n",
+    "    image_hdu: 1\n",
+    "    badpix_hdu: 2\n",
+    "    weight_hdu: 3\n",
+    "        \n",
+    "    # Define the input catalog.\n",
+    "    cat_file_name: D00572501_z_c01_r5473p01_piff.fits\n",
+    "        \n",
+    "    # Either (x,y) or (ra,dec) are required columns saying where the objects are.\n",
+    "    x_col: XWIN_IMAGE\n",
+    "    y_col: YWIN_IMAGE\n",
+    "        \n",
+    "    # If flags is set, it will ignore flagged rows.\n",
+    "    flag_col: FLAGS\n",
+    "        \n",
+    "    # You can optionally specify any other columns you want to read.\n",
+    "    # These will be saved as additional properties of each star, which may be used in various ways.\n",
+    "    property_cols: [SPREAD_MODEL]\n",
+    "        \n",
+    "    # This sets how large a stamp to pull out of the full image for each input object.\n",
+    "    stamp_size: 25\n",
+    "\n",
+    "\n",
+    "select:\n",
+    "    # The default is to use all input objects as candidate stars.\n",
+    "    # Here we refine that a bit by selecting rows based on the SExtractor \"SPREAD_MODEL\" value.\n",
+    "    type: Properties\n",
+    "        \n",
+    "    # This can be any eval-able string.\n",
+    "    # Here we use the SPREAD_MODEL property we read in from the input catalog.\n",
+    "    where: \"np.abs(SPREAD_MODEL) < 3.e-4\"\n",
+    "        \n",
+    "    # Mark 20% of stars as \"reserve\" stars.  Don't use them for fitting.                     \n",
+    "    # In ML lingo, this would be a validation set.\n",
```

Comment:
Since this is a pedagogical tutorial, it'd be nice to define all acronyms like "ML", "Sextractor", and "DES" the first time they're used.